.circleci/config.yml

```diff
@@ -167,47 +166,52 @@ commands:
           key: << parameters.key_prefix >>-{{ .Branch }}
 
   manage_api_cache:
-    description: "workbench/api: save, restore or save_and_restore gradle cache"
+    description: "workbench/api: save or restore gradle cache"
     parameters:
       save:
         type: boolean
         default: false
       restore:
         type: boolean
         default: false
-      key_prefix:
-        type: string
-        default: v1-api-cache-{{ checksum "~/workbench/api/build.gradle" }}
     steps:
       - when:
           condition: << parameters.save >>
           steps:
+            # splitting into multiple distinct caches. Refers to https://circleci.com/docs/2.0/caching/#cache-size
             - save_cache:
                 paths:
-                  - ~/.gradle/caches
                   - ~/.gradle/wrapper
-                  - ~/.m2
-                  - ~/workbench/api/build/exploded-api/WEB-INF/lib/
-                key: << parameters.key_prefix >>-{{ .Branch }}
+                key: v1-gradle-wrapper-{{ checksum "~/workbench/api/gradle/wrapper/gradle-wrapper.properties" }}-{{ .Branch }}
+            - save_cache:
+                paths:
+                  - ~/.gradle/caches
+                key: v1-gradle-cache-{{ checksum "~/workbench/api/build.gradle" }}-{{ .Branch }}
       - when:
           condition: << parameters.restore >>
           steps:
             - restore_cache:
                 keys:
-                  - << parameters.key_prefix >>-{{ .Branch }}
-                  - << parameters.key_prefix >>-master
-                  - << parameters.key_prefix >>
+                  - v1-gradle-wrapper-{{ checksum "~/workbench/api/gradle/wrapper/gradle-wrapper.properties" }}-{{ .Branch }}
+                  - v1-gradle-wrapper-{{ checksum "~/workbench/api/gradle/wrapper/gradle-wrapper.properties" }}-master
+            - restore_cache:
+                keys:
+                  - v1-gradle-cache-{{ checksum "~/workbench/api/build.gradle" }}-{{ .Branch }}
+                  - v1-gradle-cache-{{ checksum "~/workbench/api/build.gradle" }}-master
 
   run_e2e_test:
     description: "Run puppeteer e2e integration tests"
     steps:
+      - run:
+          name: Update Puppeteer test environment variables
+          command: |
+            echo 'export USER_NAME=$PUPPETEER_TEST_USER' >> $BASH_ENV
+            echo 'export PASSWORD=$PUPPETEER_TEST_USER_PASSWORD' >> $BASH_ENV
+            echo 'export INVITATION_KEY=$PUPPETEER_TEST_REGISTRATION_KEY' >> $BASH_ENV
+            source $BASH_ENV
```

Comment:
Bug:
Still need this way to get env. variables to work. :(

Reply:
I looked into this on one of your other PRs, and it seemed like the nicer syntax won't work in your situation. I can't remember details though.

Reply:
Thank you, your help and review is greatly appreciated.